paper-server/src/main/java/org/bukkit/craftbukkit/block/CraftBlockEntityState.java

```diff
@@ -50,7 +50,8 @@ public CraftBlockEntityState(World world, T tileEntity) {
             if (thr instanceof ThreadDeath) {
                 throw (ThreadDeath)thr;
             }
-            throw new RuntimeException("Failed to read BlockState at: world: " + this.getWorld().getName() + " location: (" + this.getX() + ", " + this.getY() + ", " + this.getZ() + ")", thr);
+            String worldName = world == null ? "null" : world.getName();
+            throw new RuntimeException("Failed to read BlockState at: world: " + worldName + " location: (" + this.getX() + ", " + this.getY() + ", " + this.getZ() + ")", thr);
```

Comment:
No need to show the location at all if it's not placed in the world. The purpose of our change to make it have the world/position is for debugging in-world issues. Not important for unplaced states.

Reply:
Changed it